receiver/vcenterreceiver/metrics.go

```diff
@@ -37,6 +37,21 @@ func (v *vcenterMetricScraper) recordVMUsages(
 	vm mo.VirtualMachine,
 	hs mo.HostSystem,
 ) {
+	diskUsed := vm.Summary.Storage.Committed
+	diskFree := vm.Summary.Storage.Uncommitted
+
+	v.mb.RecordVcenterVMDiskUsageDataPoint(now, diskUsed, metadata.AttributeDiskStateUsed)
+	v.mb.RecordVcenterVMDiskUsageDataPoint(now, diskFree, metadata.AttributeDiskStateAvailable)
+
+	if vm.Config.Template {
```

Comment:
Out of ignorance, why do we want to record `used` and `available`, but not `utilization` for a VM template?

Reply:
@crobert-1 the more I think about it...the current state of available metrics for VM templates is not the greatest.

`used` I believe represents how much space the template is currently taking up.
`available` is actually kind of meaningless for a template though. `used` + `available` would = `provisioned`. I believe this number would be the amount of space that is provisioned for any VMs that are created for this template. I don't "think" this number is actually provisioned for the template itself...but I'm not 100% sure of that.

So ultimately `used` has value. `available` only has value in adding it to `used` to see the total. And `utilization` doesn't really make sense for multiple reasons. For a template, `used`/`available` aren't really dynamic values (as far as I can tell). And the total used to calculate `utilization` doesn't really make sense in the context of the template and the space the template takes up, but more for the VMs that would be spawned from the template.

Does that make sense?

Reply:
If I'm understanding properly, `used` is definitely valuable, `available` _might_ be valuable, so it's being included just in case. Is that fair? If so I'm fine with this logic. I assume it's obvious to users familiar with vcenter that this metric wouldn't be available for templates? If it's not obvious, should we document it?

Reply:
@crobert-1 These are good questions. I think it's pretty obvious that any sort of "utilization" (CPU, Disk, Memory) metric of any kind doesn't make sense for VM templates. Only "provisioned" type metrics (but we don't currently track these for memory/cpu) and current disk space usage for the template itself make sense. So I think we should be ok without extra documentation.